include/rocksdb/metrics_reporter.h

```diff
@@ -38,6 +38,35 @@ class LatencyHistGuard {
   decltype(std::chrono::high_resolution_clock::now()) begin_time_;
 };
 
+class LatencyHistLoggedGuard {
+ public:
+  explicit LatencyHistLoggedGuard(HistReporterHandle* handle, Logger* logger , unsigned  int threshold, const char* tag = "LatencyHistLoggedGuard")
+      : handle_(handle),
+        begin_time_(std::chrono::high_resolution_clock::now()),
+        log_threshold_us_(threshold),
+        logger_(logger),
+        tag_(tag){}
+
+  ~LatencyHistLoggedGuard() {
+    if (handle_ != nullptr) {
+      auto us = std::chrono::duration_cast<std::chrono::microseconds>(
+          std::chrono::high_resolution_clock::now() - begin_time_)
+          .count();
+      handle_->AddRecord(us);
+      if(us > log_threshold_us_){
```

Comment:
Please run `clang-format` on this fiilie. btw, what about also printing out a stacktrace?

Reply:
done